File: conciertosBack/api/src/main/java/com/rubenazo/buscaConciertos/adapters/in/dto/SalaConciertoDto.java
package com.rubenazo.buscaConciertos.adapters.in.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.rubenazo.buscaConciertos.domain.SalaConcierto;

public record SalaConciertoDto(
    String id,
    String name,
    String address,
    String city,
    String province,
    Double lat,
    Double lng,
    @JsonProperty("image_url") String imageUrl,
    String description,
    @JsonProperty("source_url") String sourceUrl
) {
    public static SalaConciertoDto from(SalaConcierto sala) {
        return new SalaConciertoDto(
            sala.id(),
            sala.name(),
            sala.address(),
            sala.city(),
            sala.province(),
            sala.lat(),
            sala.lng(),
            sala.imageUrl(),
            sala.description(),
            sala.sourceUrl()
        );
    }
}

File: conciertosFront/shared/src/commonMain/kotlin/org/rubenazo/conciertosfront/core/data/remote/dto/SalaConciertoDto.kt
package org.rubenazo.conciertosfront.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SalaConciertoDto(
    @SerialName("id") val id: String,
    @SerialName("name") val name: String,
    @SerialName("address") val address: String?,
    @SerialName("city") val city: String,
    @SerialName("province") val province: String,
    @SerialName("lat") val lat: Double?,
    @SerialName("lng") val lng: Double?,
    @SerialName("image_url") val imageUrl: String?,
    @SerialName("description") val description: String?,
    @SerialName("source_url") val sourceUrl: String?
)
